airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeDestinationHandler.kt

```diff
@@ -536,42 +537,52 @@ class SnowflakeDestinationHandler(
-                |SELECT table_schema, table_name, column_name, data_type, is_nullable 
-                |FROM information_schema.columns 
-                |WHERE table_catalog = ? 
-                |AND table_schema IN (${IntRange(1, streamIds.size).joinToString { "?" }}) 
-                |AND table_name IN (${IntRange(1, streamIds.size).joinToString { "?" }}) 
-                |ORDER BY table_schema, table_name, ordinal_position; 
-                |""".trimMargin()
-
-            val bindValues =
-                arrayOf(databaseName.uppercase(Locale.getDefault())) + namespaces + names
-            val results: List<JsonNode> = database.queryJsons(query, *bindValues)
-            for (result in results) {
-                val tableSchema = result["TABLE_SCHEMA"].asText()
-                val tableName = result["TABLE_NAME"].asText()
-                val columnName = result["COLUMN_NAME"].asText()
-                val dataType = result["DATA_TYPE"].asText()
-                val isNullable = result["IS_NULLABLE"].asText()
-                val tableDefinition =
+        ): Map<String, Map<String, TableDefinition>> {
+            val existingTables = HashMap<String, HashMap<String, TableDefinition>>()
+            for (stream in streamIds) {
+                val schemaName = stream.finalNamespace
+                val tableName = stream.finalName
+                val table = getTable(database, schemaName, tableName)
+                if (table != null) {
                     existingTables
-                        .computeIfAbsent(tableSchema) { _: String? -> LinkedHashMap() }
-                        .computeIfAbsent(tableName) { _: String? ->
-                            TableDefinition(LinkedHashMap())
-                        }
-                tableDefinition.columns[columnName] =
-                    ColumnDefinition(columnName, dataType, 0, fromIsNullableIsoString(isNullable))
+                        .computeIfAbsent(schemaName) { _: String? -> HashMap() }
+                        .computeIfAbsent(tableName) { _: String? -> table }
+                }
             }
             return existingTables
         }
+
+        fun getTable(
+            database: JdbcDatabase,
+            schemaName: String,
+            tableName: String,
+        ): TableDefinition? {
+            try {
+                val columns = LinkedHashMap<String, ColumnDefinition>()
+                database.queryJsons("DESCRIBE TABLE \"$schemaName\".\"$tableName\"").map {
+                    val columnName = it["name"].asText()
+                    val dataType =
+                        when (
+                            val snowflakeDataType =
+                                it["type"].asText().takeWhile { char -> char != '(' }
+                        ) {
+                            "VARCHAR" -> "TEXT"
+                            else -> snowflakeDataType
+                        }
+
+                    val isNullable = it["null?"].asText() == "Y"
```

Comment:
this continues to cause me physical pain 😅 (nothing we can do about it, just feels dumb)